aten/src/ATen/core/boxing/KernelFunction_impl.h

```diff
@@ -56,6 +70,9 @@ inline Return KernelFunction::callUnboxed(const OperatorHandle& opHandle, Args..
     // forwarding, which would require Args to be deduced, but instead we
     // want callers to explicitly specify the Args.
 
+    TORCH_INTERNAL_ASSERT(!signature_hash_.has_value() || (detail::hashFunctionSignature<Return (Args...)>() == *signature_hash_),
+      "Called KernelFunction::callUnboxed with wrong argument types");
```

Comment:
Aren't you adding a big honking check in the hotpath for the code?

Reply:
It *should* be a fast check since it only compares two ints (the `hashFunctionSignature` is constexpr). I'll measure it before landing though.

Reply:
Yes, and  all of the intrusive pointer asserts *should* be fast because they're just boolean tests, and look where that got us!